### ISSUE TITLE
DataTable with fixedLayout, fixedHeader and selectRows="checkbox" generates warnings in console

### ISSUE THREAD

**Issue by hannes-o-brunner** [salesforce/design-system-react#2383]
@salesforce/design-system-react@0.10.12

A DataTable with fixedLayout, fixedHeader and selectRows="checkbox" generates warnings in the console, like

```
index.js:1375 Warning: Failed prop type: Invalid prop `name` of type `boolean` supplied to `SLDSCheckbox`, expected `string`.
    in SLDSCheckbox (at head.jsx:136)
    in SLDSDataTableHead (at data-table/index.jsx:475)
    in table (at data-table/index.jsx:451)
    in div (at data-table/index.jsx:558)
    in div (at data-table/index.jsx:544)
    in SLDSDataTable (at Report.tsx:100)
```

I assume this is eventually due to the following code in line 144 in components/data-table/private/head.jsx
`name={!ariaHidden && 'SelectAll'}`
which should rather be something like 
`name={ariaHidden ? "" : 'SelectAll'}`

**Comment by hannes-o-brunner** [comment]
There are two more places with the same pattern, which could need a fix as well (IMO)

components/data-table/private/head.jsx, line 184
`fixedHeader` && 'ariaHidden'

components/global-header/dropdown.jsx, line 44
`iconSize={globalAction` && 'small'}

**Comment by hannes-o-brunner** [comment]
Happy to provide the fix, btw

**Comment by hannes-o-brunner** [comment]
Has been fixed with #2398
